Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs

```diff
@@ -1244,34 +998,450 @@ private void ShowFilterElementsPage()
+
+                await HighlightSelectedElement(pointOnImg);
+
+                xUCElementDetails.SelectedElement = currentHighlightedElement;
+
+                mCurrentControlTreeViewItem = WindowExplorerCommon.GetTreeViewItemForElementInfo(currentHighlightedElement);
+
+                ShowCurrentControlInfo();
+
+                LastSearchFinished = true;
+            }
+        }
+
+        private ElementInfo mCurrentHighlightedElement;
+        public ElementInfo currentHighlightedElement
+        {
+            get
+            {
+                return mCurrentHighlightedElement;
+            }
+            set
+            {
+                mCurrentHighlightedElement = value;
+            }
+        }
+
+        private async Task HighlightSelectedElement(System.Windows.Point pointOnImg)
+        {
+            try
+            {
+                //calculate clicked point on mobile
+                System.Windows.Point pointOnAppScreen = GetPointOnAppWindow(pointOnImg);
+                long pointOnMobile_X = (long)pointOnAppScreen.X;
+                long pointOnMobile_Y = (long)pointOnAppScreen.Y;
+
+                //get the clicked element
+                ElementInfo inspectElementInfo = await ((IVisualTestingDriver)mApplicationAgent.Agent.Driver).GetElementAtPoint(pointOnMobile_X, pointOnMobile_Y);
+                if (inspectElementInfo != null && inspectElementInfo != currentHighlightedElement)
+                {
+                    //mark the element bounds on image
+                    DrawElementRectangleAsync(inspectElementInfo);
+
+                    //TODO: select the node in the xml
+                    currentHighlightedElement = inspectElementInfo;
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToUser(eUserMsgKey.MobileShowElementDetailsFailed, ex.Message);
+            }
+        }
+
+        private void DrawElementRectangleAsync(ElementInfo clickedElementNode)
+        {
+            try
+            {
+                //remove previous rectangle
+                RemoveElemntRectangle();
+
+                //rectangleXmlNode = clickedElementNode;
+
+                //get the element location
+                double ratio_X = 0;
+                double ratio_Y = 0;
+                double element_Start_X = clickedElementNode.X;
+                double element_Start_Y = clickedElementNode.Y;
+                double element_Max_X = clickedElementNode.X + clickedElementNode.Width;
+                double element_Max_Y = clickedElementNode.Y + clickedElementNode.Height;
+
+                //draw the rectangle
+                //mHighightRectangle = new System.Windows.Shapes.Rectangle();
+                double rectangleWidth = -1, rectangleHeight = -1;
+
+                /// Mobile specific calculations
+                if (mWindowExplorerDriver is GenericAppiumDriver)
```

Comment:
why calculations been done here? this is platform/driver specific and should be placed there, WindowExplorer page should contain Generic code only

Reply:
because it involves the AppWindow for calculating the point on image so in case we move it to drivers, it would require working with an instance of ScreenShotViewPage to be sent as parameter